File: MB-Secretaria/confidence_py/multimodal_conflict.py
from .types import ClinicalConfidenceInput, MultimodalConflictResult


def detect_multimodal_conflict(item: ClinicalConfidenceInput) -> MultimodalConflictResult:
    conflicts: list[str] = []
    if item.modality == "text":
        return MultimodalConflictResult(False, conflicts)
    if item.image_summary:
        conflicts.extend([f"image:{value}" for value in item.image_summary.conflicts])
    conflicts.extend([f"retrieval:{value}" for value in item.retrieval_summary.conflicts])
    conflicts.extend([f"memory:{value}" for value in item.memory_context_summary.conflicts])
    if item.modality == "multimodal" and not (item.image_summary and item.image_summary.available):
        conflicts.append("multimodal_image_evidence_missing")
    return MultimodalConflictResult(bool(conflicts), conflicts)

File: MetaBrain/confidence/multimodal-conflict.ts
import type { ClinicalConfidenceInput, MultimodalConflictResult } from "./types";

export function detectMultimodalConflict(input: ClinicalConfidenceInput): MultimodalConflictResult {
  const conflicts: string[] = [];

  if (input.modality === "text") return { multimodal_conflict_detected: false, conflicts };
  if (input.image_summary?.conflicts?.length) conflicts.push(...input.image_summary.conflicts.map((item) => `image:${item}`));
  if (input.retrieval_summary.conflicts?.length) conflicts.push(...input.retrieval_summary.conflicts.map((item) => `retrieval:${item}`));
  if (input.memory_context_summary.conflicts?.length) conflicts.push(...input.memory_context_summary.conflicts.map((item) => `memory:${item}`));
  if (input.modality === "multimodal" && !input.image_summary?.available) conflicts.push("multimodal_image_evidence_missing");

  return {
    multimodal_conflict_detected: conflicts.length > 0,
    conflicts,
  };
}
